terraform/prow_clusters/prow_clusters.tf

```diff
@@ -0,0 +1,149 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+# Multi repo standard clusters
+module "multi-repo-stable" {
+  source = "../testgroup"
+  channel = "stable"
+}
+
+module "multi-repo-regular" {
+  source = "../testgroup"
+  channel = "regular"
+}
+
+module "multi-repo-rapid" {
+  source = "../testgroup"
+  channel = "rapid"
+}
+
+module "multi-repo-rapid-latest" {
+  source = "../testgroup"
+  channel = "rapid"
+  min_master_version = "latest"
+}
+
+module "multi-repo-psp" {
+  source = "../testgroup"
+  channel = "regular"
+  suffix = "psp"
+}
+
+module "multi-repo-bitbucket" {
+  source = "../testgroup"
+  channel = "regular"
+  suffix = "bitbucket"
+}
+
+module "multi-repo-gitlab" {
+  source = "../testgroup"
+  channel = "regular"
+  suffix = "gitlab"
+}
+
+# Multi repo autopilot clusters
+module "multi-repo-autopilot-stable" {
+  source = "../testgroup_autopilot"
+  channel = "stable"
+}
+
+module "multi-repo-autopilot-regular" {
+  source = "../testgroup_autopilot"
+  channel = "regular"
+}
+
+module "multi-repo-autopilot-rapid" {
+  source = "../testgroup_autopilot"
+  channel = "rapid"
+}
+
+module "multi-repo-autopilot-rapid-latest" {
+  source = "../testgroup_autopilot"
+  channel = "rapid"
+  min_master_version = "latest"
+}
+
+# Mono repo clusters
+module "mono-repo-stable" {
+  source = "../testgroup"
+  prefix = "mono-repo"
+  channel = "stable"
+  num_clusters = 3
+}
+
+module "mono-repo-regular" {
+  source = "../testgroup"
+  prefix = "mono-repo"
+  channel = "regular"
+  num_clusters = 3
+}
+
+module "mono-repo-rapid" {
+  source = "../testgroup"
+  prefix = "mono-repo"
+  channel = "rapid"
+  num_clusters = 3
+}
+
+module "mono-repo-rapid-latest" {
+  source = "../testgroup"
+  prefix = "mono-repo"
+  channel = "rapid"
+  min_master_version = "latest"
+  num_clusters = 3
+}
+
+# One off clusters
+module "multi-repo-kind" {
+  source = "../e2ecluster"
+  name = "multi-repo-kind"
+  channel = "regular"
+  subnetwork = google_compute_subnetwork.e2e-subnetwork-1.name
+  network = google_compute_network.e2e-network.name
```

Comment:
All other multi-repo and mono-repo clusters use the default network. Are we going to hit any quota issue?

Reply:
The constraint I was trying to workaround here is the maximum number of secondary IP ranges in each subnetwork (maximum is 30). Note that we are creating many more clusters now with all of the test groups, which requires more subnetworks to accommodate them.

I chose to just create a new network for each test group to make it simpler to manage the CIDR ranges in each subnetwork - otherwise we need to do some kind of bookkeeping on CIDR ranges to make sure there is no overlap between subnetworks. Alternative is to create multiple subnetworks in a single network (e.g. default) and handle the bookkeeping for all of the CIDR ranges. WDYT?

Reply:
To answer your question though there are some quotas that I had to increase in my testing. I successfully managed to provision all of the prow resources in a dev project to verify. 
- CPU quota (to accommodate the large number of GKE clusters)
- Firewall quota (to accommodate the large number of VPC firewall rules required for the large number of GKE clusters)

Reply:
Cool. I'm good with a new network for each test group as long as no more quota issue.

We may need to regroup these tests or create more test groups in the future for more features. If we go with bookkeeping for CIDR ranges, I guess, it is subject to change.